src/blocks/SliderNew/utils.ts

```diff
@@ -47,3 +52,198 @@ export function getSliderResponsiveParams({
         return res;
     }, {} as Record<number, SwiperOptions>);
 }
+
+export const getSlideId = (sliderId: string, index: number) => `slider-${sliderId}-child-${index}`;
+
+const MEMOIZABLE_TYPES = ['object', 'function'];
+export const useMemoized = <T>(value: T): T => {
+    const [memoizedValue, setMemoizedValue] = useState(value);
+
+    useEffect(() => {
+        setMemoizedValue((memoized) =>
+            typeof value in MEMOIZABLE_TYPES && isEqual(memoized, value) ? memoized : value,
+        );
+    }, [value]);
+
+    return memoizedValue;
+};
+
+interface PaginationBulletData {
```

Comment:
Is it possible/useful to use `bullets` prop from `{PaginationMethods} from 'swiper/types'`?

Reply:
It is useful to access the DOM elements, though it is typed wrongly as Dom7Array[]. It is used inside usePaginationBullets() to form PaginationBulletData. This interface keeps the bullets together with their event handlers for the latter to be removed after rerenders